src/AuditTrailModule/javasource/test_crm/tests/TestAuditInheritance.java

```diff
@@ -56,22 +58,166 @@ public void testCreatedRecord() throws InterruptedException, CoreException {
 		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.CompanyNr.toString(), "", COMPANY_NR));
 		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.InternNr.toString(), "", company.getInternNr().toString()));
 		expectedLog.addLogLine(new ExpectedLogLineReferenceSet(this.context, Company.MemberNames.Company_Group.toString()));
-		
+
+		final ActualLog actualLog = new ActualLog(context, company.getMendixObject().getId().toLong());
+		expectedLog.verify(actualLog);
+	}
+
+	@Test
+	public void testChangeRecord() throws InterruptedException, CoreException {
+		final Date initialDate = new Date();
+		// Create new record
+		final Company company = createBaseCompany();
+
+		company.setName(NAME2);
+		company.commit();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = new ExpectedLog(1, TypeOfLog.Change, Company.entityName, adminId, initialDate, Company.entityName);
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.Name.toString(), NAME, NAME2));
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.CompanyNr.toString(), COMPANY_NR, COMPANY_NR));
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.InternNr.toString(), company.getInternNr().toString(), company.getInternNr().toString()));
+		expectedLog.addLogLine(new ExpectedLogLineReferenceSet(this.context, Company.MemberNames.Company_Group.toString()));
+
+		final ActualLog actualLog = new ActualLog(context, company.getMendixObject().getId().toLong());
+		expectedLog.verify(actualLog);
+	}
+
+	@Test
+	public void testDeleteRecord() throws InterruptedException, CoreException {
+		final Date initialDate = new Date();
+		// Create new record
+		final Company company = createBaseCompany();
+
+		company.delete();
+
+		// Assert log was created
+		final ExpectedLog expectedLog = new ExpectedLog(0, TypeOfLog.Delete, Company.entityName, adminId, initialDate, Company.entityName);
+		expectedLog.addLogLine(new ExpectedLogLineAttribute(Company.MemberNames.Name.toString(), NAME, NAME));
```

Comment:
I see some kind of repetition in the creation of an `expectedLog` across different tests. It might be an idea to extract a reusable method to reduce the verbosity here and use them across different tests.

Reply:
New code should handle that. See if you like it! :)